### PR TITLE
Contour circularity

### PR DESCRIPTION
**Pull request by wagenrace**
Thank you for your contribution to the **Tomni** package. 
Before submitting/accepting this Pull Request, please make sure:

## Contributor of code
- [x] Functions are named based on [Python naming conventions](https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html)
- [x] Explain how to run the code `(docstring)`
- [x] Function is tested
- [x] All needed package are in [requirements.txt](requirements.txt)
- [x] Requirements in setup.py are updated
- [x] Version number is updated
- [x] `HISTORY.rst` is updated
- [x] New feature is added to `README.md` subsection `Features`

## Reviewer
- [x] Functions are named based on [Python naming conventions](https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html)
- [x] The code is executed on another machine
- [x] Function is tested
- [x] Installing just the [requirements.txt](requirements.txt) is enough to run the code
- [x] Requirements in setup.py are updated
- [x] Version number is updated
- [x] `HISTORY.rst` is updated
- [x] New feature is added to `README.md` subsection `Features`

### CODE REVIEW

File: tomni/contour_operations/circularity/main.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import cv2
+import numpy as np
+
+
+def circularity(cnt: np.ndarray) -> float:
+    """
+    Calculates the circularity of a contour
```
Comment (schneiderjan) [inline tomni/contour_operations/circularity/main.py:7]: ... contour.

File: tomni/contour_operations/circularity/main.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    This is based on the perimeter and area of the contour.
+    A circle will give the smallest area for a given perimeter, so this is 1
```
Comment (schneiderjan) [inline tomni/contour_operations/circularity/main.py:9]: ... is 1.

File: tomni/contour_operations/circularity/main.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    The smaller the area is for a perimeter the less circular the contour is.
+    So the closer is get to 0
```
Comment (schneiderjan) [inline tomni/contour_operations/circularity/main.py:12]: ... to 0.

File: tomni/contour_operations/circularity/main.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    The formula works from calculating the r**2 based on Area and Perimeter
+    Area based:
+    r**2 = A/pi
+
+    Perimeter based:
+    r**2 = (p/2*pi)**2
+
+    divinding these two gives circularity
```
Comment (schneiderjan) [inline tomni/contour_operations/circularity/main.py:21]: Dividing

File: tomni/contour_operations/circularity/main.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    c = (A/pi) / (p/2*pi)**2 = (4 * pi * A) / (p**2)
+
+    Args:
+        cnt (np.ndarray): Contour as given be opencv
```
Comment (schneiderjan) [inline tomni/contour_operations/circularity/main.py:25]: by.

File: tomni/contour_operations/circularity/test_circularity.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
+from unittest import TestCase
+import numpy as np
+from .main import circularity
+
+
+class TestCircularity(TestCase):
+    def test_perfect_square(self):
+        contour = np.array([[[500, 50]], [[500, 99]], [[549, 99]], [[549, 50]]])
+
+        expectedResult = np.pi / 4
```
Comment (schneiderjan) [inline tomni/contour_operations/circularity/test_circularity.py:10]: should be `expected_result`

**Comment by schneiderjan** [comment]
Add new feature to readme.

File: tomni/contour_operations/circularity/main.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+    The formula works from calculating the r**2 based on Area and Perimeter
+    Area based:
+    r**2 = A/pi
+
+    Perimeter based:
+    r**2 = (p/2*pi)**2
+
+    Dividing these two gives circularity
+    c = (A/pi) / (p/2*pi)**2 = (4 * pi * A) / (p**2)
```
Comment (jellevankerk) [inline tomni/contour_operations/circularity/main.py:22]: is this based on some source. if so it would be good to add the source in the doc string

Comment (wagenrace) [reply tomni/contour_operations/circularity/main.py:22]: Found it in some documents but never a source...

Comment (jellevankerk) [reply tomni/contour_operations/circularity/main.py:22]: what kind of documents, is it possible to refer to them?

**Pull request merged**